lib/src/main/java/com/diffplug/spotless/npm/PrettierFormatterStep.java

```diff
@@ -104,4 +104,41 @@ private void endServer(PrettierRestService restService, ServerProcessInfo restSe
 		}
 
 	}
+
+	public static class PrettierFilePathPassingFormatterFunc implements FormatterFunc {
+		private final String prettierConfigOptions;
+		private final PrettierRestService restService;
+
+		public PrettierFilePathPassingFormatterFunc(String prettierConfigOptions, PrettierRestService restService) {
+			this.prettierConfigOptions = requireNonNull(prettierConfigOptions);
+			this.restService = requireNonNull(restService);
+		}
+
+		@Override
+		public String apply(String input) throws Exception {
+			throw new IllegalStateException("should never be called");
```

Comment:
It is still possible that someone will call `apply(String input, File null)`.  So if you want to throw an error if the user doesn't pass you a real file, the way to do that is not to override this method, but to throw whatever error you want in `apply(String, File)`.

Reply:
Alright. But I'll have to override the method none the less, since the `apply(String, File)` overload does delegate to `apply(String)` and not the other way around.

Reply:
>  I'll have to override the method none the less

I forgot, and you are correct.  My main point is that `File` can be null, and the two `apply` methods ought to behave 
the same for that case, rather than throwing "should never be called" in one path, and NPE in the other.